app/controllers/api/v1/users/sessions_controller.rb

```diff
@@ -8,6 +8,22 @@ def create
     end
   end
 
+  def destroy
+    # fetch refresh token from request header
+    refresh_token = request.headers["Authorization"]&.split(" ")&.last
+    # find user's api credentials by refresh token
+    api_credential = ApiCredential.find_by(refresh_token_digest: Digest::SHA256.hexdigest(refresh_token))
```

Comment:
nit: api_credential = ApiCredential is redundant instead do.

```
if ApiCredential.find_by(refresh_token_digest: Digest::SHA256.hexdigest(refresh_token))
  api_credential.revoke_api_token
  api_credential.revoke_refresh_token
  render json: {message: "Signed out successfully."}, status: 200
else
  ...

Reply:
I see, traditionally though, it's the `access token` that goes in the header on every response. 

This goes beyond just sign-out and will reflect the shape of our responses from the client past sign-in for every request so should scope this.

But if we're going to sign the user out based on just one token value it should be the `access token` because that value is valid for less time/more secure.

Reply:
Ah I see. That makes sense now.
Then let's put the access token in the header and use that for sign out.